modules/hostap/src/supp_api.c

```diff
@@ -2082,6 +2082,78 @@ int hapd_state(const struct device *dev, int *state)
 	return ret;
 }
 
+static int hapd_config_chan_center_seg0(struct wifi_connect_req_params *params)
+{
+	int ret = 0;
+	uint8_t center_freq_seg0_idx = 0;
+	uint8_t oper_chwidth = CHANWIDTH_USE_HT;
+	uint8_t *center_freq = NULL;
+	uint8_t center_freq_40MHz[] = {38, 46, 54, 62, 102, 110, 118, 126, 134, 142, 151, 159};
+	uint8_t center_freq_80MHz[] = {42, 58, 106, 122, 138, 155};
+	uint8_t index, index_max, chan_idx, ch_offset = 0;
+
+	/* Unless ACS is being used, both "channel" and "vht_oper_centr_freq_seg0_idx"
+	 * parameters must be set. */
+	switch (params->bandwidth) {
+	case WIFI_FREQ_BANDWIDTH_20MHZ:
+		oper_chwidth = CHANWIDTH_USE_HT;
+		center_freq_seg0_idx = params->channel;
+		break;
+	case WIFI_FREQ_BANDWIDTH_40MHZ:
+		if (!hostapd_cli_cmd_v("set ht_capab [HT40+][SHORT-GI-40]")) {
+			goto out;
+		}
+		oper_chwidth = CHANWIDTH_USE_HT;
+		center_freq = center_freq_40MHz;
+		index_max = ARRAY_SIZE(center_freq_40MHz);
+		ch_offset = 2;
+		break;
+	case WIFI_FREQ_BANDWIDTH_80MHZ:
+		if (!hostapd_cli_cmd_v("set ht_capab [HT40+][SHORT-GI-40]")) {
+			goto out;
+		}
+		oper_chwidth = CHANWIDTH_80MHZ;
+		center_freq = center_freq_80MHz;
+		index_max = ARRAY_SIZE(center_freq_80MHz);
+		ch_offset = 6;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (params->bandwidth != WIFI_FREQ_BANDWIDTH_20MHZ)
+	{
+		chan_idx = params->channel;
+		for (index = 0; index < index_max; index++)
+		{
+			if ((chan_idx >= (center_freq[index] - ch_offset)) && (chan_idx <= (center_freq[index] + ch_offset)))
+			{
+					center_freq_seg0_idx = center_freq[index];
+					break;
+			}
+		}
+	}
```

Comment:
can't we just do below?

```
	if (params->bandwidth != WIFI_FREQ_BANDWIDTH_20MHZ)
	{
		center_freq_seg0_idx = params->channel + ch_offset;
	}
```
We don't even need the static array of frequencies? Of course `+/-` depends on the HT40+/- user input.

Reply:
Yes, I will check it.